accounts-db/src/append_vec.rs

```diff
@@ -728,9 +728,16 @@ impl AppendVec {
             }
             AppendVecFileBacking::File(file) => {
                 // 4096 was just picked to be a single page size
-                let mut buf = [0u8; PAGE_SIZE as usize];
-                let bytes_read = read_into_buffer(file, self.len(), offset, &mut buf).ok()?;
-                let valid_bytes = ValidSlice(&buf[..bytes_read]);
+                let mut buf = [MaybeUninit::<u8>::uninit(); PAGE_SIZE as usize];
+                // SAFETY: `read_into_buffer` will only write to uninitialized memory.
+                let bytes_read = read_into_buffer(file, self.len(), offset, unsafe {
+                    std::slice::from_raw_parts_mut(buf.as_mut_ptr() as *mut u8, PAGE_SIZE as usize)
```

Comment:
In theory just creating `&mut [T]` here triggers UB, since writing over existing `T`
triggers `T::drop`. From the docs:

```
data must point to len consecutive properly initialized values of type T.
```

However, since currently `u8::drop()` does nothing, _and_ `file.read_at()` feeds the
pointer straight to pread() and that is unlikely to ever change, I think this is a
case where the perf improvement justifies relying on the compiler/std impl. 

Nothing to do here, just recording this for posterity.

Reply:
nit: use buf.len() instead of PAGE_SIZE for the len?

Reply:
Miri agrees that this code is indeed not UB. Though one can never trust miri.